starter-code/scripts/articleView.js

```diff
@@ -73,41 +73,69 @@ articleView.setTeasers = function() {
+// The articleView.handleMainNav function is doing this for us already, so...
+// Time spent: 20 minutes figuring out that nothing was needed...
 
-  $('#article-json').on('focus', function(){
+  // DONE: Hide the article-export section on page load
+  $('#article-export').hide();
+  $('#article-json').on('focus', function(e){
+    e.preventDefault();
     this.select();
   });
 
-  // TODO: Add an event handler to update the preview and the article-export field if any inputs change.
-
-};
+  // DONE: Add an event handler to update the preview and the article-export field if any inputs change.
+  articleView.create();
+}
 
 // this is the function that generates the preview and shows the export field
 articleView.create = function() {
-  // TODO: Set up a var to hold the new article we are creating.
-  // Clear out the #articles element, so we can put in the updated preview
-
-
-  // TODO: Instantiate an article based on what's in the form fields:
-
-
-  // TODO: Use our interface to the Handblebars template to put the article preview into the DOM:
-
-
-  // TODO: The new articles we create will be shown as JSON in an element in our article-export section. From there, we can copy/paste the JSON into our source data file.
+  $('#entryForm').on('change', function(event){
+    event.preventDefault();
+    // Clear out the #articles element, so we can put in the updated preview
+    $('#articles').empty();
+
+    // DONE: Set up a var to hold the new article we are creating.
+    var newEntry = new Entry(this);
+    // DONE: Instantiate an article based on what's in the form fields:
+    // DONE: Use our interface to the Handblebars template to put the article preview into the DOM:
+    //we call the `Entry` constructor, then pass in `this` which is referencing the form
+    //that we've placed the eventlistener on and we assign the returned obj to 'newEntry' var
+    articleView.templateAndRender(newEntry);
+    $('#articles').show();
+
+    // DONE: The new articles we create will be shown as JSON in an element in our article-export section. From there, we can copy/paste the JSON into our source data file.
     // Set up this "export" functionality. When data is inputted into the form, that data should be converted to stringified JSON. Then, display that JSON in the element inside the article-export section. The article-export section was hidden on page load; make sure to show it as soon as data is entered in the form.
+    var jsonStr = JSON.stringify(newEntry);
+    $('#article-json').empty();
+    $('#article-json').append(jsonStr);
+    $('#article-export').show();
 
+  });
 };
 
-
 articleView.initIndexPage = function() {
   articleView.populateFilters();
   articleView.handleCategoryFilter();
   articleView.handleAuthorFilter();
   articleView.handleMainNav();
   articleView.setTeasers();
+  articleView.initNewArticlePage();
```

Comment:
Why would you call this here? This is initIndexPage. But we are working on new.html. Where do we ask you to call a function in this assignment?